include/kernel/sporadic.h

```diff
@@ -119,6 +119,15 @@ static inline bool_t refill_ready(sched_context_t *sc)
     return refill_head(sc)->rTime <= (NODE_STATE_ON_CORE(ksCurTime, sc->scCore) + getKernelWcetTicks());
 }
 
+/*
+ * Return true if an SC has been successfully configured with parameters
+ * that allow for a thread to run.
+ */
+static inline bool_t sc_active(sched_context_t *sc)
```

Comment:
sc_active isn't a clear name to me as it could be mistaken about what it refers to. As  `sc->scRefillMax > 0` is used in many places to check that the SC has been configured after retyping, and isn't in the middle of being revoked maybe something like sc_initialized() to more clearly indicate that if the predicate is false then the SC attributes have invalid values?

Reply:
Also, if adding sc_active is to help code understandability, then I think we should also update the other occurrences of (sc->scRefillMax > 0) so that there is still a consistent way of checking this.

Reply:
This name was chosen to reflect what was being used in the proofs. If we are seeing that check elsewhere in the code then it would make sense to use it consistently. `sc_initialised` would make more sense, I'd have to see what such a change would mean for the proof team; I don't want to add more work than I have to on their side but I suspect a change like this wouldn't impact the abstract spec but would be good to get in before more changes to the haskell or refine are made.

Reply:
This name was originally devised for the proofs before there was a need to name it in the kernel code. I could try to change it but that might mean some changes in the proofs or some oddities in refine down the road.